### PR TITLE
doc: fill out `fs` and remove html links

### PR DESCRIPTION
**Pull request by carllerche**
Switch to rustdoc path links instead of relative HTML links.

This also fills out `tokio::fs` API documentation.

Work towards #1473

### CODE REVIEW

File: tokio/src/fs/open_options.rs [hunk 1]
```diff
@@ -32,62 +92,279 @@ impl OpenOptions {
...
+    ///
+    /// This is an async version of [`std::fs::OpenOptions::truncate`][std]
     ///
-    /// [`truncate`]: https://doc.rust-lang.org/std/fs/struct.OpenOptions.html#method.truncate
+    /// [std]: std::fs::OpenOptions::truncate
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use std::fs::OpenOptions;
```
Comment (LucioFranco) [inline tokio/src/fs/open_options.rs]: shouldn't this be tokio?

File: tokio/src/process/mod.rs [hunk 2]
```diff
@@ -346,7 +346,7 @@ impl Command {
     /// whether it should be interpreted relative to the parent's working
     /// directory or relative to `current_dir`. The behavior in this case is
     /// platform specific and unstable, and it's recommended to use
-    /// [`canonicalize`] to get an absolute program path instead.
+    /// `canonicalize` to get an absolute program path instead.
```
Comment (ipetkov) [inline tokio/src/process/mod.rs]: If there's no `tokio` equivalent of this function we should link to the `std` one

Comment (carllerche) [reply tokio/src/process/mod.rs]: I didn't look into whether or not we needed one yet.

Comment (carllerche) [reply tokio/src/process/mod.rs]: I believe we do need a version.

Comment (carllerche) [reply tokio/src/process/mod.rs]: Ok, I added it and fixed the link.

File: tokio/src/fs/open_options.rs (hunk 1 above)
Comment (carllerche) [reply tokio/src/fs/open_options.rs]: thanks :+1:

**Pull request merged**